x/ecocredit/client/marketplace/query.go

```diff
@@ -0,0 +1,242 @@
+func QuerySellOrdersCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sell-orders",
+		Short: "List all sell orders with pagination",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, err := sdkclient.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			client := marketplace.NewQueryClient(ctx)
+			pagination, err := sdkclient.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+			res, err := client.SellOrders(cmd.Context(), &marketplace.QuerySellOrdersRequest{
+				Pagination: pagination,
+			})
+			if err != nil {
+				return err
+			}
+
+			return ctx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "sell-orders")
+
+	return cmd
+}
+
+// QuerySellOrdersByAddressCmd returns a query command that retrieves all sell orders by address with pagination.
+func QuerySellOrdersByAddressCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sell-orders-by-address [address]",
+		Short: "List all sell orders by owner address with pagination",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, err := sdkclient.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			client := marketplace.NewQueryClient(ctx)
+			pagination, err := sdkclient.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+			res, err := client.SellOrdersByAddress(cmd.Context(), &marketplace.QuerySellOrdersByAddressRequest{
+				Address:    args[0],
+				Pagination: pagination,
+			})
+			if err != nil {
+				return err
+			}
+
+			return ctx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "sell-orders-by-address")
+
+	return cmd
+}
+
+// QuerySellOrdersByBatchDenomCmd returns a query command that retrieves all sell orders by batch denom with pagination.
+func QuerySellOrdersByBatchDenomCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sell-orders-by-batch-denom [batch_denom]",
+		Short: "List all sell orders by batch denom with pagination",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, err := sdkclient.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			client := marketplace.NewQueryClient(ctx)
+			pagination, err := sdkclient.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+			res, err := client.SellOrdersByBatchDenom(cmd.Context(), &marketplace.QuerySellOrdersByBatchDenomRequest{
+				BatchDenom: args[0],
+				Pagination: pagination,
+			})
+			if err != nil {
+				return err
+			}
+
+			return ctx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "sell-orders-by-batch-denom")
+	return cmd
+}
+
+// QueryBuyOrderCmd returns a query command that retrieves information for a given buy order.
+func QueryBuyOrderCmd() *cobra.Command {
```

Comment:
i dont think we actually need the buy order queries anymore (this one and the 2 below). buy orders are being removed for v4.0 and will likely be reintroduced later. ref: #967

i also have a branch ready to introduce direct-buy-order CLI cmd once #967 is merged

Reply:
Oof. Apologies for the extra work migrating the buy order commands. These can be removed but we can use the work you did here when we are ready to reimplement.